infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -455,8 +483,10 @@ private Timer getTimer(byte reqType) {
     private void handleRawToken(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
         final TokenRequest req = msg.getPayload();
 
-        Token token = new Token(sequencerEpoch, globalLogTail.getAndAdd(req.getNumTokens()));
-        r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(new TokenResponse(token, Collections.emptyMap())));
+        Token token = new Token(sequencerEpoch, globalLogTail);
```

Comment:
Maybe add a comment about how the tail points to an open slot and not the last written slot and therefore you return the current position and then update.

Reply:
Done.